### PR TITLE
support different tasks in graph construction.

### PR DESCRIPTION
**Pull request by zheng-da**
*Description of changes:*
We need to support more tasks in graph construction.
https://github.com/awslabs/graphstorm/issues/51

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: python/graphstorm/gconstruct/construct_graph.py [hunk 1]
```diff
@@ -336,37 +356,74 @@ def process_labels(data, label_confs):
...
+        if valid_label_idx is None:
+            rand_idx = np.random.permutation(num_samples)
+        else:
+            rand_idx = np.random.permutation(valid_label_idx)
+        num_labels = len(rand_idx)
+        train_idx = rand_idx[0:int(num_labels * train_split)]
+        val_start = int(num_labels * train_split)
+        val_end = int(num_labels * (train_split + val_split))
         val_idx = rand_idx[val_start:val_end]
         test_idx = rand_idx[val_end:]
```
Comment (zhjwy9343) [inline python/graphstorm/gconstruct/construct_graph.py:411]: So here test_split does not take effect. We assume the rest will all become test set?

Comment (zhjwy9343) [inline python/graphstorm/gconstruct/construct_graph.py:411]: ditto here

Comment (zheng-da) [reply python/graphstorm/gconstruct/construct_graph.py:411]: this is to split the data with valid labels. What should we do with the rest of labelled samples?

Comment (zhjwy9343) [reply python/graphstorm/gconstruct/construct_graph.py:411]: I mean if users give [0.5, 0.1, 0.1]. It means 50% for train, 10% for val, and 10% for test. There will be 30% unused. Base on the current logic. The test set will be 40%, which is not users want.

Comment (zheng-da) [reply python/graphstorm/gconstruct/construct_graph.py:411]: i see. i'll fix it.

**Review by zhjwy9343 — APPROVED** [review]
LGTM

**Pull request merged**